### DISCUSSION TITLE
[Announcement][HW4] segmentation fault when instrumenting json parser

### DISCUSSION THREAD

**Discussion by KihongHeo** [prosyslab-classroom/cs348-information-security#472] [Announcements]
There is a problem when your sanitizer instruments the json parser.
Please do the followings:

- Change `is_debug` to 
```ocaml
 match Llvm.instr_opcode instr with
 | Llvm.Opcode.Call ->
     let callee_expr = Llvm.operand instr (Llvm.num_operands instr - 1) in
     let r1 = Str.regexp "llvm\\.dbg\\..+" in
     Str.string_match r1 (Llvm.value_name callee_expr) 0
 | _ -> false
```
- Change your sanitizer not to instrument `PHI` node. 

If you checkout after my fix, you will get the fixed `is_debug` code. For other students who have already checked out the code, please change the above by yourself.

**Comment by BaurTokenov** [comment]
I am not instrumenting PHI nodes, but I am still getting this error: "PHI nodes not grouped at top of the basic block!". The only Opcode I'm looking at is SDiv. Does anyone know what could be the reason?

**Comment by gunsodo** [comment]
@BaurTokenov This issue also occurs with me. In my case, I have to specifically ignore PHI nodes as what we have done with debug in the previous assignment. You may want to use some function in `utils.ml` that might help.
